core/src/main/java/org/jboss/pnc/reqour/service/scmcreation/GitHubApiService.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2024 Red Hat, Inc.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.jboss.pnc.reqour.service.scmcreation;
+
+import java.io.IOException;
+
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Inject;
+
+import org.jboss.pnc.api.enums.InternalSCMCreationStatus;
+import org.jboss.pnc.reqour.common.exceptions.GitHubApiException;
+import org.jboss.pnc.reqour.config.ConfigConstants;
+import org.jboss.pnc.reqour.config.ConfigUtils;
+import org.jboss.pnc.reqour.config.GitProviderConfig;
+import org.jboss.pnc.reqour.config.GitProviderFaultTolerancePolicy;
+import org.jboss.pnc.reqour.model.GitHubProjectCreationResult;
+import org.jboss.pnc.reqour.runtime.UserLogger;
+import org.kohsuke.github.GHOrganization;
+import org.kohsuke.github.GHRepository;
+import org.kohsuke.github.GitHub;
+import org.slf4j.Logger;
+
+import io.quarkus.arc.lookup.LookupIfProperty;
+import io.smallrye.faulttolerance.api.ApplyGuard;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * Thin wrapper around {@link GitHub} for performing GitHub API calls.
+ */
+@ApplicationScoped
+@LookupIfProperty(name = ConfigConstants.ACTIVE_GIT_PROVIDER, stringValue = ConfigConstants.GITHUB)
+@Slf4j
+public class GitHubApiService {
+
+    private final GitHub delegate;
+    private final GitProviderConfig gitProviderConfig;
+    private final Logger userLogger;
+
+    @Inject
+    public GitHubApiService(GitHub delegate, ConfigUtils configUtils, @UserLogger Logger userLogger) {
+        this.delegate = delegate;
+        gitProviderConfig = configUtils.getActiveGitProviderConfig();
+        this.userLogger = userLogger;
+    }
+
+    /**
+     * Gets a repository given by its repository name from the GitHub's internal organization.
```

Comment:
Returns the repository if it exists but more importantly, it creates one if it doesn't exist.